lib/tools/apk-utils.js

```diff
@@ -198,6 +198,47 @@ apkUtilsMethods.install = async function (apk, replace = true, timeout = 60000)
   }
 };
 
+apkUtilsMethods.installOrUpgrade = async function (apk, pkg = null, timeout = 60000) {
+  let apkInfo = null;
+  if (!pkg) {
+    apkInfo = await this.getApkInfo(apk);
+    pkg = apkInfo.name;
+  }
+  if (!pkg) {
+    log.warn(`Cannot read the package name of ${apk}. Assuming correct app version is already installed`);
+    return;
+  }
+  if (!await this.isAppInstalled(pkg)) {
+    await this.adbExec(['install', apk], {timeout});
```

Comment:
Seems like we could just `await this.install(apk, false, timeout);`. Then if installing changed sometime down the line it would only be in one place.